daemon/api_aspects.go

```diff
@@ -0,0 +1,118 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+/*
+ * Copyright (C) 2023 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package daemon
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/snapcore/snapd/aspects"
+	"github.com/snapcore/snapd/overlord/auth"
+)
+
+var (
+	aspectsCmd = &Command{
+		Path:        "/v2/aspects",
```

Comment:
can we explore having the API for now be like:

```
GET /v2/aspects/<account-id>/<bundle>/<aspect-name>?fields=<dotted-field1>,<dotted-field2>
PUT  /v2/aspects/<account-id>/<bundle>/<aspect-name>
{
  "dotted-field1": ...,
  "dotted-field2": ...
}
```
This is closer to what we do in api_snap_conf.go.

Similarly I wonder if the PUT case should return an async result even if we don't need that for now.

I also wonder

Reply:
Ok, I'll change it to use the query parameter. Is this just to be similar to the config API or are there other reasons?

> Similarly I wonder if the PUT case should return an async result even if we don't need that for now.

I thought we could change as it was needed since this is anyway experimental but I can do it now, if you prefer

Reply:
mostly to make it obvious that we can implement the snap set/get commands without too many changes in them

Reply:
I see, thanks